programs/margin-account/src/lib.rs

```diff
@@ -190,15 +190,28 @@ pub mod margin_account {
         Ok(())
     }
     /// Withdraw funds from an obligation account.
-    pub fn withdraw(_ctx: Context<Withdraw>) -> ProgramResult {
-        // TODO
-        Ok(())
-    }
-    /// Liquidate a position if below liquidation price.
-    //? This potentially isn't needed, if the logic can happen on the lending pool (obligation
-    //? account) but for now it's assumed the transaction will have to go through here.
-    pub fn liquidate(_ctx: Context<Liquidate>) -> ProgramResult {
-        // TODO
+    pub fn withdraw(ctx: Context<Withdraw>, amount: u64) -> ProgramResult {
+        if amount == 0 {
+            return Err(ErrorCode::InvalidAmount.into());
+        };
+        let margin_account = &mut ctx.accounts.margin_account;
+        let position = margin_account
+            .position
+            .as_mut()
```

Comment:
```suggestion
            .as_ref()
```
You don't need it to be mutable